frontend/gateway/gateway.go

```diff
@@ -87,6 +100,30 @@ func filterPrefix(opts map[string]string, pfx string) map[string]string {
 	return m
 }
 
+func (gf *gatewayFrontend) checkSourceIsAllowed(source string) error {
+	// Returns nil if the source is allowed.
+	// Returns an error if the source is not allowed.
+	if len(gf.allowedSources) == 0 {
+		// No source restrictions in place
+		return nil
+	}
+
+	sourceRef, err := reference.ParseNormalizedNamed(source)
+	if err != nil {
+		return err
+	}
+
+	taglessSource := reference.TrimNamed(sourceRef).Name()
```

Comment:
If the match is purely based on repo name then I guess we should call the option "allowedRepositories". The wildcard concept also makes `alpine`, `library/alpine` and `docker.io/library/alpine` all different. If there were no wildcards then we could normalize the references to same format and do direct comparison.